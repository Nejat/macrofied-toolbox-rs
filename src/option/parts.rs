bitflags! {
    pub struct Parts: u8 {
        const NOTHING = 0b00000;
        const SOME    = 0b00001;
        const DEBUG   = 0b00010;
        const NONE    = 0b00100;
        const SOME_DEBUG      = Self::SOME.bits  | Self::DEBUG.bits;
        const SOME_NONE       = Self::SOME.bits  | Self::NONE.bits;
        const SOME_DEBUG_NONE = Self::SOME.bits  | Self::DEBUG.bits     | Self::NONE.bits;
        const DEBUG_NONE      = Self::DEBUG.bits | Self::NONE.bits;
    }
}
