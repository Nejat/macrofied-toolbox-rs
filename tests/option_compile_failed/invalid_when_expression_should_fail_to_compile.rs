use macrofied_toolbox::option;

fn main() {
    option! {
        @when foo();
        @some "will not compile";
    }
}

fn foo() -> usize {
    42
}