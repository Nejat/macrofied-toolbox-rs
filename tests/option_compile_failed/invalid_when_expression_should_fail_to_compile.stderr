error[E0599]: no method named `is_some` found for type `usize` in the current scope
 --> tests/option_compile_failed/invalid_when_expression_should_fail_to_compile.rs:4:5
  |
4 | /     option! {
5 | |         @when foo();
6 | |         @some "will not compile";
7 | |     }
  | |_____^ method not found in `usize`
  |
  = note: this error originates in the macro `option` (in Nightly builds, run with -Z macro-backtrace for more info)
